[Week1] 1주차 과제

usuuhyn opened the pull request:
## 📌 PR 요약

🫛 작업한 내용
- 회원가입 (SignUpActivity)
   > ID, 비밀번호, 닉네임, 주량 입력 UI
   > 유효성 검사 추가
   > 회원가입 성공 시 입력 데이터 Intent로 반환
- 로그인 (SignInActivity)
   > 회원가입에서 전달 받은 데이터 저장 및 검증
   > 로그인 성공 시 메인 화면으로 이동
- 메인 (MainActivity)
   > 로그인 시 전달받은 사용자 정보 표시

- 공통 컴포넌트
  > `Button`, `TextField`, `Title`, `InfoItem`
___

🫛 PR 포인트
- registerForActivityResult와 Intent를 활용한 화면 간 데이터 전달 구조
- Composable 재사용성을 위한 공통 컴포넌트화

## 📸 스크린샷
<video src="https://github.com/user-attachments/assets/ae99b1a3-a42e-44f4-bbb1-40669db6f47e" /> 



## 📮 관련 이슈
- closed #2

usuuhyn committed:
feat: 로그인 화면 UI 구현

- SignInActivity.kt 추가
- LabeledTextField 컴포저블 추가
- 비밀번호 입력 시 *** 처리
- TextField placeholder 스타일 수정
- 버튼 색상 및 모양 커스터마이징

usuuhyn committed:
refactor: SignButton 및 Title 컴포저블 분리

usuuhyn committed:
feat: 회원가입 화면 UI 구현

- SignUpActivity.kt 추가
- LabeledTextField, Title, SignButton 재사용

usuuhyn committed:
refactor: 재사용 가능한 UI 컴포넌트를  component  폴더로 분리

usuuhyn committed:
chore: 초기 실행 화면을 SignInActivity로 변경

usuuhyn committed:
feat: 회원가입 결과 전달 및 로그인 로직 구현

- registerForActivitiyResult를 사용해 SignUpActivity에서 회원가입 정보 반환
- SignInActivity에서 회원가입 결과를 받아 로그인 검증 기능 추가

usuuhyn committed:
chore: 불필요한 import 및 중복 코드 제거

usuuhyn committed:
feat: 메인 화면 구현 및 데이터 표시

- MainActivity.kt
- Intent로 전달된 값 수신
- 공통 컴포넌트 InfoItem 생성 및 재사용
- 로그인 성공 시 putExtra로 값들 전달하도록 연결

kimjw2003 commented on app/src/main/java/com/sopt/dive/screen/MainActivity.kt:
```diff
@@ -0,0 +1,113 @@
+        super.onCreate(savedInstanceState)
+
+        // 로그인 화면에서 받은 데이터
+        val id = intent.getStringExtra("아이디") ?: ""
```
key는 영어로 해주시면 좋을것같아요! 또한 Const로 빼주시는 것도 좋은 방법이 될것 같아요

kimjw2003 commented on app/src/main/java/com/sopt/dive/screen/SignInActivity.kt:
```diff
@@ -0,0 +1,140 @@
+    ) {
+        // 타이틀
+        Title(
+            text = "Welcome To SOPT"
```
앱에서 사용되는 텍스트들은 strings.xml에서 일괄 관리해주면 좋을것 같아요

sssuhha replied:
넵 !!

kimjw2003 reviewed (approved):
수고하셨습니다~

seungjunGong commented on app/src/main/java/com/sopt/dive/component/InfoItem.kt:
```diff
@@ -0,0 +1,36 @@
+fun InfoItem(
+    title: String,
+    value: String,
+    modifier: Modifier = Modifier,
```
굿굿 !!

seungjunGong commented on app/src/main/java/com/sopt/dive/component/TextField.kt:
```diff
@@ -0,0 +1,51 @@
+        TextField(
+            value = text,
+            onValueChange = onValueChange,
+            placeholder = { Text(placeholder, color = Color.Gray, fontSize = 15.sp)},
```
해당 컴포넌트를 한줄로 넣어주신 이유가 있나요? 개인 선호도의 차이긴합니다

sssuhha replied:
줄바꿈....습관이 안됏기 때문입니다.. 허허 이것도 수정하겠습니다!

seungjunGong commented on app/src/main/java/com/sopt/dive/screen/MainActivity.kt:
```diff
@@ -0,0 +1,113 @@
+            // 프로필 이미지
+            Image(
+                painter = painterResource(id = R.drawable.profile),
+                contentDescription = "profile",
```
contentDescription 이 하는 역할은 무엇일까요?

seungjunGong commented on app/src/main/java/com/sopt/dive/screen/MainActivity.kt:
```diff
@@ -0,0 +1,113 @@
+    ) {
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
```
지금 Modifier 에 `fillMaxSize()` 를 전부 설정해주셨는데 해당 함수는 화면의 전체 영역을 가지게 하기 때문에 최상단 컴포넌트에만 해주시면 됩니다. 여기서는 맨위의 Column 에만 지정해주시면 돼요 ~

sssuhha replied:
넵!!

seungjunGong commented on app/src/main/java/com/sopt/dive/screen/MainActivity.kt:
```diff
@@ -0,0 +1,113 @@
+        nickname: String,
+        etc: String
+) {
+    Column (
```
지금 아래 컬럼이랑 중복이 되기 때문에 하나는 지워도 됩니다.
레이아웃의 depth 를 깊게 가져가는 것은 성능상 좋지 않아요

seungjunGong commented on app/src/main/java/com/sopt/dive/screen/SignInActivity.kt:
```diff
@@ -0,0 +1,140 @@
+
+    // 사용자 입력값
+    var id by remember { mutableStateOf("") }
+    var password by remember { mutableStateOf("")}
```
여기서 질문 ! 사용자가 데이터를 입력한 후 화면전환과 같은 구성변경이 일어나면 값이 어떻게 될까요?
한번 텍스트 필드 입력 후 화면 전환을 해보시고 값이 변경된다면 이를 해결할 수 있는 방법을 찾아보시면 좋을거 같아요

seungjunGong commented on app/src/main/java/com/sopt/dive/screen/SignInActivity.kt:
```diff
@@ -0,0 +1,140 @@
+}
+
+@Composable
+fun SignInScreen() {
```
여기두 private !

seungjunGong reviewed (approved):
구현 깔끔하게 잘해주셨네요 ~! 리뷰 보시고 수정할 부분이 있다면 반영해주심 좋을거 같습니다 !
앞으로도 같이 파이팅해봅시다 🙌

vahkjsdf reviewed (approved):
코드 잘봤습니당 가독성 좋게 잘 짜주셔서 읽는데 편했어요!! 앞으로 잘 부탁드려요~

sssuhha commented on app/src/main/java/com/sopt/dive/screen/MainActivity.kt:
```diff
@@ -0,0 +1,113 @@
+}
+
+@Composable
+fun MyPageScreen( // 읽기 전용 -> 인자로 전달 받기
```
넵!!

usuuhyn committed:
[#2/chore] 줄바꿈 및 들여쓰기 수정

usuuhyn committed:
[#2/refactor] 로직과 UI를 분리하여 구조 개선

usuuhyn committed:
[#2/refactor] 문자열 리소스 및 Intent key 상수화로 코드 구조 개선

- 하드코딩된 텍스트 -> strings.xml 로 분리
- IntentKeys 객체 추가하여 Intent key를 상수로 관리

usuuhyn committed:
[#2/refactor] Activity 내부 전용 Screen을 private으로 변경

usuuhyn committed:
[#2/refactor] PasswordVisualTransformation 재사용하도록 최적화

- PasswordVisualTransformation을 remember로 감싸 불필요한 객체 생성 방지
- 리컴포지션 시 동일 인스턴스 재사용으로 성능 개선

usuuhyn committed:
[#2/refactor] 불필요한 Column 및 fillMaxSize 속성 제거

usuuhyn committed:
[#2/refactor] rememberSaveable로 상태 유지하도록 수정

usuuhyn committed:
[#2/chore] edge-to-edge 적용으로 시스템 바 영역 대응

usuuhyn committed:
[#2/refactor] 회원가입 검증 로직 분리

- etc 항목은 빈칸 허용 조건으로 처리

usuuhyn committed:
[#2/fix] 로그인 검증 조건에 registeredId.isNotBlank() 추가
